src/EditorFeatures/Core/Extensibility/NavigationBar/AbstractEditorNavigationBarItemService.cs

```diff
@@ -26,15 +28,16 @@ protected AbstractEditorNavigationBarItemService(IThreadingContext threadingCont
         protected abstract Task<VirtualTreePoint?> GetSymbolNavigationPointAsync(Document document, ISymbol symbol, CancellationToken cancellationToken);
         protected abstract Task<bool> TryNavigateToItemAsync(Document document, WrappedNavigationBarItem item, ITextView textView, CancellationToken cancellationToken);
 
-        public async Task<IList<NavigationBarItem>?> GetItemsAsync(Document document, CancellationToken cancellationToken)
+        public async Task<ImmutableArray<NavigationBarItem>> GetItemsAsync(
+            Document document, ITextSnapshot textSnapshot, CancellationToken cancellationToken)
```

Comment:
note: this is the *editor* portion of the API.  it does pass along the textsnaphsot now as the *Editor* portion returns navbaritems that have tracking spans in them.  so this allows us to create those tracking spans at construction time, not some point in the future after.  the *feature* api is of course oblivious to this.